src/Guardrail.sol

```diff
@@ -222,18 +230,29 @@ contract Guardrail is ITransactionGuard, IModuleGuard, MultiSendCallOnlyv2 {
         bytes calldata,
         address
     ) external override {
-        bytes4 selector = _decodeSelector(data);
-        if (to == msg.sender && selector == GuardManager.setGuard.selector) {
+        _checkOperationAndAllowance(msg.sender, to, _decodeSelector(data), operation);
+    }
+
+    function _checkAfterExecution() internal {
+        address guard = abi.decode(SafeInterface(msg.sender).getStorageAt(uint256(GUARD_STORAGE_SLOT), 1), (address));
+        address moduleGuard =
+            abi.decode(SafeInterface(msg.sender).getStorageAt(uint256(MODULE_GUARD_STORAGE_SLOT), 1), (address));
+
+        if (guard != address(this) && moduleGuard != address(this)) {
             _removeGuard(msg.sender);
+        } else if (guard != address(this) || moduleGuard != address(this)) {
+            // This error is thrown if only one of the guard is removed.
```

Comment:
Ideally, the comments should explain why, not what, the code already explains what is happening :D
Something like "we should not allow the situation where the guard is enabled only as a module guard OR a regular guard"